### ISSUE TITLE
Not able to assign resources that is cpu request and limits and memory request and limits for operator pod containers.

### ISSUE THREAD

**Issue by AnilKumarBandi27** [operator-framework/operator-lifecycle-manager#2204]
I have added the resources that is cpu requests and limits and memory requests and limits to both the operator containers as shown in below CSV file. After the operator installation resources section is empty i.e. resources: { }.

Please find below attached CSV file operator deployment section:

```
install:
    spec:
      deployments:
      - name: <operator-name>  
        spec:
          replicas: 1
          selector:
            matchLabels:
              name: <operator-name>  
          template:
            metadata:
              annotations:
                name: <operator-name>           
              labels:
                name: <operator-name>          
            spec:
              containers:
              - name: kube-rbac-proxy
                image: gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0
                resources:
                  limits:
                    cpu: 200m
                    memory: 100Mi
                  requests:
                    cpu: 100m
                    memory: 50Mi
              - name: manager
                image: docker.io/<operator-name>@sha256:<sha-256-code>
                resources:
                  limits:
                    cpu: 900m
                    memory: 400Mi
                  requests:
                    cpu: 1500m
                    memory: 1000Mi
```

After the installation, no resources are allocated. The operator pod YAML file looks as shown below.
```
install:
    spec:
      deployments:
      - name: <operator-name>  
        spec:
          replicas: 1
          selector:
            matchLabels:
              name: <operator-name>  
          template:
            metadata:
              annotations:
                name: <operator-name>           
              labels:
                name: <operator-name>          
            spec:
              containers:
              - name: kube-rbac-proxy
                image: gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0
                resources: {}
              - name: manager
                image: docker.io/<operator-name>@sha256:<sha-256-code>
                resources: {}
```
Environment

Ansible-based operator - operator-sdk v1.7.2

Tested on OpenShift v4.5, v4.6 and  v4.7 the behavior is the same.

Expected to show the resources defined in the CSV file in operator pod YAML file after the successful installation.

**Comment by horis233** [comment]
@AnilKumarBandi27 

It is a known issue, please check this github issue: https://github.com/operator-framework/operator-lifecycle-manager/issues/1940 and bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1937375

The OLM fix is included in OCP 4.6.29, as per

https://access.redhat.com/errata/RHBA-2021:1521
https://bugzilla.redhat.com/show_bug.cgi?id=1945149#c7

And in OCP 4.7.5, as per
https://access.redhat.com/errata/RHSA-2021:1005
https://bugzilla.redhat.com/show_bug.cgi?id=1937375#c7

**Comment by exdx** [comment]
Closing as this is addressed by #2010
